helm/prometheus-rules/templates/alerting-rules/app.rules.yml

```diff
@@ -45,8 +45,11 @@ spec:
       annotations:
         description: '{{`Workload Cluster App {{ if $labels.exported_namespace }}{{ $labels.exported_namespace }}{{ else }}{{ $labels.namespace }}{{ end }}/{{ $labels.name }}, version {{ $labels.version }} is {{if $labels.status }} in {{ $labels.status }} state. {{else}} not installed. {{end}}`}}'
         opsrecipe: app-failed/
-      # we have this Frankenstein for namespace checks because go (and prometheus) don't support negative lookaheads in regex
+      {{- if eq .Values.managementCluster.provider.flavor "capi" }}
+      expr: label_replace(app_operator_app_info{status!~"(?i:(deployed|cordoned|not-installed))", catalog~="giantswarm|cluster", team!~"^$|noteam"}, "cluster_id", "$1", "name", "([a-zA-Z0-9]+)-.*") == 1
```

Comment:
just for reference, this rule will end up matching on the MC `cluster-(provider)` app and `default-apps-(provider)` as well as the WC ones so it might get confusing that it fires under `WorkloadClusterappFailed`

Reply:
I agree, I'm trying to find out how to merge the 2 but with it's hard to get the installation name as a cluster ID in vintage MC alerts

Reply:
I think we would need to app exporter to configure the cluster-id in the metric